content/news/XXXX-XX-XX-dvs-internals-pt2.md

```diff
@@ -0,0 +1,301 @@
+| 3        | `011`        |
+| 4        | `100`        |
+| 5        | `101`        |
+| 6        | `110`        |
+| 7        | `111`        |
+
+Of course, that is just a toy example, but even if we only had 8 different positions, we need use 3 bits (= 3 cycles) to represent each position.
+In reality, we want to encode *a lot* more different positions, which means the resulting binary number will be longer, meaning we need more bits and therefore cycles per position we want to encode.
+
+The corresponding signal would look like this:
+
+![Simple bitstream with 3 bits per position]({static}/images/news/timecode-3-bit-bitstream.svg)
+
+The first 3 cycles encode position 0, the next 3 cycles encode position 1, and so on.
+Simple, right?
+Unfortunately, this approach has a major problem.
+
+Let's assume you seek to a random position in the track (by picking up the needle and dropping it somewhere) and the
+next 3 bits are `001`.
+Do you know what position or cycle number we are at?
+
+We can look up `001` in the table above and see that these bits encode position 1 (cycles 3-6).
+So after reading these bits we should be at cycle 6.
+And that is *one* possibility, but not the only one.
+
+![Simple bitstream with 3 bits per position and all occurrences of 001 highlighted]({static}/images/news/timecode-3-bit-bitstream-highlight.svg)
+
+So there are multiple possible possible positions and we don't know which one is the correct one.
+We can mitigate this by reading the next 3 cycles, too, but this doubles the amount of cycles we have to read (6 instead of 3 cycles) until we know the position for sure.
+
+DVS systems are supposed to be responsive and minimize the latency (i.e. the delay between moving the vinyl and seeing/hearing the action in the software) to be usable for scratching.
+Therefore, this approach is not sufficient.
+
+The problem with the simple base-2 encoding is that may end up in the middle of a number and we don't know where a number begins or ends.
+
+### Using Sentinel Values
+
+To detect the beginning of a new position mark reliably, we need something special, that can be clearly distinguished from the position values.
+Such a value is called a [sentinel value](https://en.wikipedia.org/wiki/Sentinel_value) and could be used to indicate "A new position value starts here".
+
+After a seek, we have to keep reading bits until we encounter the sentinel value, then we can start reading the actual position bits.
+In fact, that's exactly what the early [Final Scratch DVS system](https://en.wikipedia.org/wiki/Final_Scratch#Vinyl/CD_time_code) did.
+
+In the Final Scratch timecode, each position was prefixed by `0001`, followed by 16 bit of position information.
+Although using a sentinel value ensures that we know when a position starts, this approach is also problematic.
+Let me explain why.
+
+As we already stated, we want the delay (i.e. the latency) between moving the needle and detecting the position to be as small as possible.
```

Comment:
```suggestion
As we already stated, we want the latency between moving the needle and detecting the position to be as small as possible.
```
A Delay is a time shift between an expected event ant the actual event. Since this does not fit well here I would drop the word here. The relationship is already explained well above